### ISSUE TITLE
Memory Leak in pngtest

### ISSUE THREAD

**Issue by ocean1** [glennrp/libpng#265]
[leak_30925.zip](https://github.com/glennrp/libpng/files/2713172/leak_30925.zip) causes a memory leak in pngtest detected by asan instrumentation:

```
==27580==ERROR: LeakSanitizer: detected memory leaks

Direct leak of 36 byte(s) in 3 object(s) allocated from:
    #0 0x4db940 in malloc (/libpng/.libs/pngtest+0x4db940)
    #1 0x7f30194efa4a in png_malloc_base /libpng/pngmem.c:95:17
    #2 0x7f30194ef93b in png_malloc /libpng/pngmem.c:179:10
    #3 0x51a3ff in test_one_file /libpng/pngtest.c:1436:25
    #4 0x514d4f in main /libpng/pngtest.c:2034:19
    #5 0x7f30182ecb96 in __libc_start_main /build/glibc-OTsEL5/glibc-2.27/csu/../csu/libc-start.c:310
```

```
./configure CFLAGS=-O0 -g -fsanitize=address CPPFLAGS=-O0 -g -fsanitize=address CXXFLAGS=-O0 -g -fsanitize=address --enable-static
make
```

**Comment by ocean1** [comment]
I can confirm the behavior on Ubuntu 18.04, compiled with clang 6.0: ./configure --enable-static, then manually modified the Makefile adding -fsanitize=address to CFLAGS, CPPFLAGS, and LDFLAGS. Overwriting the pngtest.png file with the testcase, then running pngtest will give the asan output pasted above. Thanks!

**Comment by ctruta** [comment]
Many thanks for your insistence :-)
Fixed in master.

**Comment by ocean1** [comment]
Thank you!

**Comment by ctruta** [comment]
Fixed in libpng-1.6.37.
